### PR TITLE
Move `raise_or_lower_index` and `trace` to Tensor lib

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Removes some library dependencies on general relativity.

### Upgrade instructions

The functions `raise_or_lower_index` and `trace` are now declared in `DataStructures/Tensor/EagerMath/`.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
Can't fix the clang-tidy issue in SpecExporter.hpp (since SpEC is not installed on CI).

File: tests/Unit/DataStructures/Tensor/EagerMath/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,8 @@ set(LIBRARY_SOURCES
   Test_Magnitude.cpp
   Test_Norms.cpp
   Test_OrthonormalOneform.cpp
+  Test_OuterProduct.cpp
```
Comment (kidder) [inline tests/Unit/DataStructures/Tensor/EagerMath/CMakeLists.txt:14]: This wasn't being built before?  Should be in a separate commit

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/SetPiAndPhiFromConstraints.cpp [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #include "DataStructures/DataBox/Prefixes.hpp"
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/SetPiAndPhiFromConstraints.cpp]: remove

File: src/PointwiseFunctions/AnalyticData/Xcts/CommonVariables.tpp [hunk 3]
```diff
@@ -26,6 +26,7 @@
 #include "Utilities/ErrorHandling/Assert.hpp"
 #include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/Gsl.hpp"
+#include "DataStructures/Tensor/EagerMath/Trace.hpp"
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticData/Xcts/CommonVariables.tpp]: alphabet

File: src/PointwiseFunctions/AnalyticSolutions/Xcts/WrappedGr.cpp [hunk 4]
```diff
@@ -9,6 +9,7 @@
 #include "DataStructures/DataBox/Prefixes.hpp"
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/Xcts/WrappedGr.cpp]: remove

File: src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/ExtrinsicCurvature.hpp [hunk 5]
```diff
@@ -9,6 +9,7 @@
 #include "DataStructures/DataBox/Tag.hpp"
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/ExtrinsicCurvature.hpp]: remove

File: src/PointwiseFunctions/GeneralRelativity/Ricci.cpp [hunk 6]
```diff
@@ -4,6 +4,7 @@
 #include "PointwiseFunctions/GeneralRelativity/Ricci.hpp"
 
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline src/PointwiseFunctions/GeneralRelativity/Ricci.cpp]: remove

File: src/PointwiseFunctions/GeneralRelativity/Surfaces/RicciScalar.cpp [hunk 7]
```diff
@@ -7,6 +7,7 @@
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
+#include "DataStructures/Tensor/EagerMath/Trace.hpp"
```
Comment (kidder) [inline src/PointwiseFunctions/GeneralRelativity/Surfaces/RicciScalar.cpp:9]: remove

File: tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp [hunk 8]
```diff
@@ -16,6 +16,7 @@
 #include "DataStructures/DataVector.hpp"        // IWYU pragma: keep
 #include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp]: remove

File: tests/Unit/Helpers/PointwiseFunctions/AnalyticSolutions/Xcts/VerifySolution.hpp [hunk 9]
```diff
@@ -4,6 +4,7 @@
 #pragma once
 
 #include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
```
Comment (kidder) [inline tests/Unit/Helpers/PointwiseFunctions/AnalyticSolutions/Xcts/VerifySolution.hpp]: remove

**Review by kidder — CHANGES_REQUESTED** [review]
Please provide a fixup commit

File: tests/Unit/DataStructures/Tensor/EagerMath/CMakeLists.txt (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/DataStructures/Tensor/EagerMath/CMakeLists.txt:14]: No apparently is wasn't 🤷‍♂️ I just noticed and added it

**Comment by nilsvu** [comment]
done

**Review by kidder — COMMENTED** [review]
you can squash

**Comment by nilsvu** [comment]
Done

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
ignoring clang-tidy failures due to missing SpEC in CI and random test failure
